fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3451,9 +3451,9 @@ private void handleLoanRepaymentInFull(final LocalDate transactionDate, final Lo
             }
         }
         if (isAllChargesPaid) {
+            this.closedOnDate = transactionDate;
```

Comment:
Why did you move the date? If the reason was to make this date be part of the event, please move the actualMaturityDate as well to be consistent.

Reply:
I need to move the set of the `closedOnDate` value before the Life cycle machine method call because when the event listener was called It was in null at that moment

Reply:
Pleae move the actualMaturityDate as well then...

Reply:
What is weird...even the loan got changed later, since it is the same instance, the updated field should be reflected there as well.

Reply:
Done